### ISSUE TITLE
Problem with Performance/Detect

### ISSUE THREAD

**Issue by bquorning** [bbatsov/rubocop#1757]
On the latest master (933d084f7df1c68663db07f13e2d1a17b3e195b4), I tried this bit of code:
```rb
class A
  def find_all(arg)
    [arg]
  end
end

A.new.find_all(42).first
```

And got this offense: 

```
test.rb:7:7: C: Use find instead of find_all.first.
A.new.find_all(42).first
      ^^^^^^^^^^^^^^^^^^
```

Not all classes that define `#find_all` do also define an equivalent `#find` method. A more realistic example would be [`ActionView::Resolver`](https://github.com/rails/rails/blob/v4.2.1/actionview/lib/action_view/template/resolver.rb).

cc @palkan @rrosenblum

**Comment by bbatsov** [comment]
There's no real solution for this, but I doubt it's a big issue in practice. We wanted to address `#find_all` from Enumerable and we can't account for all the naming choices in 3rd party libs (Rails has so many bad ideas in that department). 

Like with any similar cop, the only thing one can in case of false positives is to disable the cop.

**Comment by bquorning** [comment]
Got it :smile:
